compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/java/KSClassDeclarationJavaImpl.kt

```diff
@@ -138,7 +138,14 @@ class KSClassDeclarationJavaImpl private constructor(val psi: PsiClass) :
     }
 
     override val superTypes: Sequence<KSTypeReference> by lazy {
-        psi.superTypes.asSequence().map { KSTypeReferenceJavaImpl.getCached(it, this) }.memoized()
+        val adjusted = if (!psi.isInterface && psi.superTypes.size > 1) {
+            psi.superTypes.filterNot {
+                it.className == "Object" && it.canonicalText == "java.lang.Object"
+            }
+        } else {
+            psi.superTypes.toList()
+        }
+        adjusted.asSequence().map { KSTypeReferenceJavaImpl.getCached(it, this) }.memoized()
```

Comment:
Will this add `Object` to super types if super type list empty?

Reply:
I doubt that the super list from psi can be empty; It seems to me that it is always available from psi, according to the test cases.